plugins/network/NetClass.cpp

```diff
@@ -51,110 +49,54 @@ void NetBrowser::LogData(const wchar_t * Data)
 }
 #endif
 
-NetResourceList::NetResourceList()
-	: ResList(),
-	ResCount(0)
-{
-}
-
-NetResourceList::~NetResourceList()
-{
-	Clear();
-}
-
-NetResourceList& NetResourceList::operator=(NetResourceList& other)
-{
-	Clear();
-
-	for (unsigned I = 0; I < other.Count(); I++)
-		Push(other[I]);
-
-	return *this;
-}
-
 void NetResourceList::Clear()
 {
-	for (unsigned i = 0; i < ResCount; i++)
-		DeleteNetResource(ResList[i]);
-
-	free(ResList);
-	ResList = {};
-	ResCount = 0;
-}
-
-void NetResourceList::DeleteNetResource(NETRESOURCE& Res)
-{
-	free(Res.lpRemoteName);
-	free(Res.lpLocalName);
-	free(Res.lpComment);
-	free(Res.lpProvider);
-}
-
-wchar_t* NetResourceList::CopyText(const wchar_t* Text)
-{
-	return Text? _wcsdup(Text) : nullptr;
+	ResList.clear();
 }
 
-void NetResourceList::CopyNetResource(NETRESOURCE& Dest, const NETRESOURCE& Src)
+void NetResourceList::Push(NetResource& Res)
 {
-	free(Dest.lpRemoteName);
-	free(Dest.lpLocalName);
-	free(Dest.lpComment);
-	free(Dest.lpProvider);
-	memcpy(&Dest, &Src, sizeof(NETRESOURCE));
-	Dest.lpRemoteName = CopyText(Src.lpRemoteName);
-	Dest.lpLocalName = CopyText(Src.lpLocalName);
-	Dest.lpComment = CopyText(Src.lpComment);
-	Dest.lpProvider = CopyText(Src.lpProvider);
+	ResList.emplace_back(Res);
```

Comment:
Res should probably be either `const NetResource&` or `NetResource&&` (and moved).